src/main/java/uk/ac/ebi/eva/pipeline/io/writers/AnnotationInVariantMongoWriter.java

```diff
@@ -0,0 +1,206 @@
+ */
+package uk.ac.ebi.eva.pipeline.io.writers;
+
+import com.mongodb.BasicDBList;
+import com.mongodb.BasicDBObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.data.MongoItemWriter;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.BasicQuery;
+import org.springframework.data.mongodb.core.query.BasicUpdate;
+import org.springframework.util.Assert;
+
+import uk.ac.ebi.eva.commons.models.data.Annotation;
+import uk.ac.ebi.eva.commons.models.data.ConsequenceType;
+import uk.ac.ebi.eva.commons.models.data.Score;
+import uk.ac.ebi.eva.commons.models.data.VariantAnnotation;
+import uk.ac.ebi.eva.commons.models.data.Xref;
+import uk.ac.ebi.eva.utils.MongoDBHelper;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.ac.ebi.eva.commons.models.data.AnnotationFieldNames.GENE_NAME_FIELD;
+import static uk.ac.ebi.eva.commons.models.data.AnnotationFieldNames.POLYPHEN_FIELD;
+import static uk.ac.ebi.eva.commons.models.data.AnnotationFieldNames.SIFT_FIELD;
+import static uk.ac.ebi.eva.commons.models.data.AnnotationFieldNames.SO_ACCESSION_FIELD;
+
+/**
+ * Update the {@link uk.ac.ebi.eva.commons.models.data.Variant} mongo document with some fields from {@link Annotation}
+ * and {@link ConsequenceType}
+ * <p>
+ * The fields are:
+ * - sifts
+ * - polyphens
+ * - soAccessions
+ * - Xref Ids
+ */
+public class AnnotationInVariantMongoWriter extends MongoItemWriter<Annotation> {
+    private static final Logger logger = LoggerFactory.getLogger(AnnotationInVariantMongoWriter.class);
+
+    private final MongoOperations mongoOperations;
+
+    private final String collection;
+
+    public AnnotationInVariantMongoWriter(MongoOperations mongoOperations, String collection) {
+        super();
+        Assert.notNull(mongoOperations, "A Mongo instance is required");
+        Assert.hasText(collection, "A collection name is required");
+
+        setCollection(collection);
+        setTemplate(mongoOperations);
+
+        this.mongoOperations = mongoOperations;
+        this.collection = collection;
+    }
+
+    @Override
+    protected void doWrite(List<? extends Annotation> annotations) {
+        Map<String, List<Annotation>> annotationsByStorageId = annotations.stream()
+                .collect(Collectors.groupingBy(MongoDBHelper::buildVariantStorageId));
+
+        for (Map.Entry<String, List<Annotation>> annotationsIdEntry : annotationsByStorageId.entrySet()) {
+            VariantAnnotation variantAnnotation = extractFieldsFromAnnotations(annotationsIdEntry.getValue());
+
+            String storageId = annotationsIdEntry.getKey();
+            BasicDBObject id = new BasicDBObject("_id", storageId);
+
+            if (mongoOperations.exists(new BasicQuery(id), collection)) {
```

Comment:
ok, now this is doing:
```
if (variant with this id exists) {
  update (addToSet) genenames
  update (addToSet) so terms
  get polyphen
  update polyphen
  get sift
  update sift
}
```
what do you think about this structure?
```
get variant with this id using a projection of only polyphen and sift
if (the variant was found) {
  DBObject update;
  update.append( addtoset gene names) // $addToSet
  update.append( addtoset so terms)
  newpolyphen = compute polyphen
  update.append ( overwrite polyphen with newpolyphen) // $set
  newsift = compute sift
  update.append ( overwrite sift with newsift)
  mongoOperations.update(update)
}
```
the first version does 3 queries and 4 updates, the second version does 1 query and 1 update